supervisor/addons/addon.py

```diff
@@ -977,11 +977,21 @@ async def install_apparmor(self) -> None:
             return
 
         # Need install/update
-        with TemporaryDirectory(dir=self.sys_config.path_tmp) as tmp_folder:
-            profile_file = Path(tmp_folder, "apparmor.txt")
+        tmp_folder: TemporaryDirectory | None = None
 
+        def install_update_profile() -> Path:
+            nonlocal tmp_folder
+            tmp_folder = TemporaryDirectory(dir=self.sys_config.path_tmp)
+            profile_file = Path(tmp_folder, "apparmor.txt")
```

Comment:
I think this should be:
```suggestion
            profile_file = Path(tmp_folder.name, "apparmor.txt")
```

(since that is what the TemporaryDirectory context manager in `__enter__` returns).